test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs

```diff
@@ -2575,6 +2576,90 @@ public void ValidateJWE(JwtTheoryData theoryData)
             TestUtilities.AssertFailIfErrors(context);
         }
 
+        // Test creates a JWT with every mapped claim and then checks that the result of validation from the
+        // JwtSecurityTokenHandler and JsonWebTokenHandler are the same, both in the mapped and unmapped case.
+        [Fact]
+        public async Task ValidateJsonWebTokenClaimMapping()
+        {
+            var jwtTokenHandler = new JsonWebTokenHandler() { MapInboundClaims = false };
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(Default.PayloadAllShortClaims),
+                SigningCredentials = KeyingMaterial.DefaultSymmetricSigningCreds_256_Sha2,
+                EncryptingCredentials = new EncryptingCredentials(KeyingMaterial.DefaultX509Key_2048, SecurityAlgorithms.RsaPKCS1, SecurityAlgorithms.Aes128CbcHmacSha256),
+            };
+
+            var accessToken = jwtTokenHandler.CreateToken(tokenDescriptor);
+
+            var validationParameters = new TokenValidationParameters
+            {
+                ValidAudience = Default.Audience,
+                ValidIssuer = Default.Issuer,
+                ValidateIssuerSigningKey = true,
+                IssuerSigningKey = KeyingMaterial.DefaultSymmetricSigningCreds_256_Sha2.Key,
+                TokenDecryptionKey = KeyingMaterial.DefaultX509Key_2048,
+                AlgorithmValidator = ValidationDelegates.AlgorithmValidatorBuilder(true),
+                RequireExpirationTime = false,
+            };
+
+            JwtSecurityTokenHandler securityTokenHandler = new JwtSecurityTokenHandler() { MapInboundClaims = false };
+
+            TokenValidationResult jwtValidationResult = await jwtTokenHandler.ValidateTokenAsync(accessToken, validationParameters);
+            TokenValidationResult securityValidationResult = await securityTokenHandler.ValidateTokenAsync(accessToken, validationParameters);
+
+            var context = new CompareContext
+            {
+                PropertiesToIgnoreWhenComparing = new Dictionary<Type, List<string>>
+                {
+                    { typeof(TokenValidationResult),  new List<string> { "SecurityToken", "TokenType" } }
+                }
+            };
+
+            Assert.True(IdentityComparer.AreEqual(jwtValidationResult, securityValidationResult, context));
```

Comment:
Instead of calling Assert.True, use [TokenUtilities.AssertFailIfErrors ](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/02518f7f8b6829be3dbb2786a0d1a644c1549aa1/test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs#L48).
Assert will throw after the first issue, when we use AssertFailIfErrors, then the test will complete with all issues.